src/conditions.lisp

```diff
@@ -47,17 +47,44 @@ PRINT-OBJECT method defined on the class named by NAME."
     `(let ((,method (find-method #'print-object '() '(,name t) nil)))
        (when ,method (remove-method #'print-object ,method)))))
 
+(defvar *in-define-condition-p* nil
+  "This will be dynamically binded to indicate an entry into DEFINE-CONDITION,
+to distinguish its DEFCLASS from an external DEFCLASS.")
+
+(defmethod ensure-class-using-class :before (class name &rest args &key direct-superclasses &allow-other-keys)
+  "In DEFINE-CONDITION: Signals a TYPE-ERROR at the first direct superclass unconforming
+with DEFINE-CONDITION, that is, the first encountered superclass in DIRECT-SUPERCLASSES
+that is not a subtype of CONDITION.
+In DEFCLASSes external to DEFINE-CONDITION: Signals an INVALID-SUPERCLASS at the first
+encountered superclass that is subtype of CONDITION."
+  (declare (ignore args))
+  (if *in-define-condition-p*
+      (loop :for superclass :in direct-superclasses
+            :do (when (not (subtypep superclass 'condition))
+                  (error 'type-error
+                         :datum superclass
+                         :expected-type 'condition)))
```

Comment:
This error is wrong - such a TYPE-ERROR would mean that the superclass is not of *type* CONDITION, whereas here the error should say that the superclass is not a *subtype* of CONDITION. Hence, I think that you should go for signaling INVALID-SUPERCLASS in here too.

Reply:
:heavy_check_mark: